web/src/OrganizationEditPage.js

```diff
@@ -112,6 +114,34 @@ class OrganizationEditPage extends React.Component {
     });
   }
 
+  updatePasswordObfuscator(key, value) {
+    const organization = this.state.organization;
+    if (organization.passwordObfuscatorType === "") {
+      organization.passwordObfuscatorType = "Plain";
+    }
+    if (key === "type") {
+      organization.passwordObfuscatorType = value;
+      organization.passwordObfuscatorKey = Obfuscator.getRandomKeyForObfuscator(value);
+      this.setState({}, () => this.passwordObfuscatorKeyRef.current?.setFields([
+        {
+          name: "passwordObfuscatorKey",
+          rules: [
+            {
+              pattern: Obfuscator.passwordObfuscatorKeyRegex[organization.passwordObfuscatorType],
+              message: `${i18next.t("organization:The password obfuscator key doesn't match the regex")}: ${Obfuscator.passwordObfuscatorKeyRegex[organization.passwordObfuscatorType]}`,
+            },
+          ],
+        },
+      ]));
+    } else if (key === "key") {
+      organization.passwordObfuscatorKey = value;
+      this.passwordObfuscatorKeyRef.current?.validateFields(["passwordObfuscatorKey"]);
```

Comment:
I just have no idea why validateFields and rules are inside a UpdateXxx() function. It doesn't make sense. Checks should be inside React Render function